arches_component_lab/src/arches_component_lab/widgets/components/GenericWidget.vue

```diff
@@ -0,0 +1,142 @@
+<script setup lang="ts">
+import { computed, ref, watchEffect } from "vue";
+
+import Message from "primevue/message";
+import Skeleton from "primevue/skeleton";
+
+import { fetchCardXNodeXWidgetData } from "@/arches_component_lab/widgets/api.ts";
+
+import {
+    CONFIGURE,
+    EDIT,
+    VIEW,
+} from "@/arches_component_lab/widgets/constants.ts";
+
+import type { CardXNodeXWidget } from "@/arches_component_lab/types.ts";
+import type { WidgetMode } from "@/arches_component_lab/widgets/types.ts";
+
+const props = withDefaults(
+    defineProps<{
+        cardXNodeXWidgetData?: CardXNodeXWidget;
+        graphSlug: string;
+        mode: WidgetMode;
+        nodeAlias: string;
+        showLabel?: boolean;
+    }>(),
+    {
+        cardXNodeXWidgetData: undefined,
+        showLabel: true,
+    },
+);
+
+const isLoading = ref(false);
+const cardXNodeXWidgetData = ref(props.cardXNodeXWidgetData);
+const configurationError = ref<Error>();
+
+const shouldShowRequiredAsterisk = computed(() => {
+    return Boolean(
+        props.mode === EDIT && cardXNodeXWidgetData.value?.node.isrequired,
+    );
+});
+
+watchEffect(async () => {
+    if (props.cardXNodeXWidgetData) {
+        return;
+    }
+
+    isLoading.value = true;
+
+    try {
+        cardXNodeXWidgetData.value = await fetchCardXNodeXWidgetData(
+            props.nodeAlias,
+            props.graphSlug,
+        );
+    } catch (error) {
+        configurationError.value = error as Error;
+    } finally {
+        isLoading.value = false;
+    }
+});
+</script>
+
+<template>
+    <div
+        class="widget"
+        :graph-slug="graphSlug"
+        :node-alias="nodeAlias"
+    >
+        <Skeleton
+            v-if="isLoading"
+            style="height: 2rem"
+        />
+        <Message
+            v-else-if="configurationError"
+            severity="error"
+            size="small"
+        >
+            {{ configurationError.message }}
+        </Message>
+        <template v-else>
+            <label
+                v-if="showLabel"
+                style="cursor: pointer; display: flex; margin-bottom: 0"
+                :for="`${graphSlug}-${nodeAlias}-input`"
```

Comment:
I think we should calc this "eleId" once and then pass that into the `:for` prop here and to the slots below rather than re-calculating it in the child components.